python/tests/kernel/test_observe_kernel.py

```diff
@@ -136,7 +136,7 @@ def kernel(theta: float, phi: float):
     assert len(energies) == 50
 
     @cudaq.kernel
-    def kernel(thetas: list):
+    def kernel(thetas: list[float]):
```

Comment:
This test is failing on Python v3.8, we can skip `list[float]` on it and have another test using `List[float]`